roc_jni/src/main/impl/receiver.c

```diff
@@ -54,35 +54,59 @@ static int receiver_config_unmarshal(JNIEnv* env, roc_receiver_config* config, j
     if (jobj != NULL)
         config->resampler_profile = (roc_resampler_profile) get_resampler_profile(env, jobj);
 
+    durationClass = (*env)->FindClass(env, "java/time/Duration");
+    assert(protocolClass != NULL);
+    getOfNanosMethodID
+            = (*env)->GetStaticMethodID(env, durationClass, "ofNanos", "(J)Ljava/time/Duration;");
+    assert(getOfNanosMethodID != NULL);
+
     // target_latency
-    config->target_latency
+    targetLatencyLong
         = get_ullong_field_value(env, receiverConfigClass, jconfig, "targetLatency", &err);
     if (err) return err;
+    config->target_latency = (jlong) (*env)->CallStaticObjectMethod(
+        env, durationClass, getOfNanosMethodID, (jlong) targetLatencyLong);
```

Comment:
here need to create from java-config C config. Looks like you do the opposite

Reply:
also keep in mind that Duration fields in java config could be null. No need to set it to C config in such case

Reply:
change